sky/provision/kubernetes/utils.py

```diff
@@ -892,6 +892,53 @@ def check_credentials(context: Optional[str],
         return True, None
 
 
+def check_pod_config(cluster_yaml_path: str, dryrun: bool) \
+    -> Tuple[bool, Optional[str]]:
+    """Check if the pod_config is a valid pod config
+
+    Using create_namespaced_pod api with dry_run to check the pod_config
+    is valid or not.
+
+    Returns:
+        bool: True if pod_config is valid.
+        str: Error message about why the pod_config is invalid, None otherwise.
+    """
+    with open(cluster_yaml_path, 'r', encoding='utf-8') as f:
+        yaml_content = f.read()
+    yaml_obj = yaml.safe_load(yaml_content)
+    pod_config = \
+        yaml_obj['available_node_types']['ray_head_default']['node_config']
+    context = yaml_obj['provider'].get('context',
+                                       get_current_kube_config_context_name())
+    try:
+        namespace = get_kube_config_context_namespace(context)
+        kubernetes.core_api(context).create_namespaced_pod(
+            namespace,
+            body=pod_config,
+            dry_run='All',
+            _request_timeout=kubernetes.API_TIMEOUT)
+    except kubernetes.api_exception() as e:
+        error_msg = ''
+        if e.body:
+            # get detail error message from api_exception
+            exception_body = json.loads(e.body)
+            error_msg = exception_body.get('message')
+        else:
+            error_msg = str(e)
+        return False, error_msg
+    except ValueError as e:
+        if dryrun:
+            logger.debug('ignore ValueError as there is no kube config '
+                         'in the enviroment with dry_run. '
+                         'For now we don\'t support check pod_config offline.')
+            return True, None
+        return False, common_utils.format_exception(e)
```

Comment:
This dry run case might not be required if we move the call to `check_pod_config` to our downstream logic in provision/kubernetes/instance.py or provision/kubernetes/config.py